corehq/apps/callcenter/sync_usercase.py

```diff
@@ -179,27 +180,30 @@ def get_sync_lock_key(user_id):
     return ["sync_user_case_for_%s" % user_id]
 
 
-def sync_call_center_user_case(user):
+def sync_call_center_user_case(user, domain):
     with CriticalSection(get_sync_lock_key(user._id)):
-        _UserCaseHelper.commit(list(_iter_call_center_case_helpers(user)))
+        domain_obj = Domain.get_by_name(domain)
+        _UserCaseHelper.commit(list(_iter_call_center_case_helpers(user, domain_obj)))
 
 
-def _iter_call_center_case_helpers(user):
-    config = user.project.call_center_config
+def _iter_call_center_case_helpers(user, domain_obj):
+    if user.is_web_user():
+        return
+    config = domain_obj.call_center_config
     if config.enabled and config.config_is_valid():
-        case, owner_id = _get_call_center_case_and_owner(user)
-        yield _get_sync_usercase_helper(user, config.case_type, owner_id, case)
+        case, owner_id = _get_call_center_case_and_owner(user, domain_obj)
+        yield _get_sync_usercase_helper(user, domain_obj.name, config.case_type, owner_id, case)
 
 
 CallCenterCaseAndOwner = namedtuple('CallCenterCaseAndOwner', 'case owner_id')
 
 
-def _get_call_center_case_and_owner(user):
+def _get_call_center_case_and_owner(user, domain_obj):
```

Comment:
Now that `_iter_call_center_case_helpers` returns early for web users, can't you remove `domain_obj` from this function and go back to using `user.domain`?